src/coreclr/interpreter/compiler.cpp

```diff
@@ -3424,6 +3424,10 @@ void InterpCompiler::EmitPushUnboxAny(const CORINFO_GENERICHANDLE_RESULT& arg1,
     PushStackType(resultStackType, clsHndStack);
     int resultVar = m_pStackPointer[-1].var;
 
+    PushStackType(StackTypeI, NULL);
+    int32_t intermediateVar = m_pStackPointer[-1].var;
+    m_pStackPointer--;
```

Comment:
I assume this is because the `intermediateVar` is temporary? I see quite a few seemingly arbitrary `m_pStackPointer--` in this file so it is the pattern, but it isn't obvious as to why it is being done in all cases. If this is indeed around creating a temporary variable, I think it would be helpful to go back in a future PR and make a helper that hides this decrement logic. For now, can someone please confirm why this is needed here and in a bunch of the places?